Plugins/Creature/Creature.cpp

```diff
@@ -1149,16 +1160,23 @@ ArgumentStack Creature::SetClericDomain(ArgumentStack&& args)
     return stack;
 }
 
-ArgumentStack Creature::GetWizardSpecialization(ArgumentStack&& args)
+ArgumentStack Creature::GetSpecialization(ArgumentStack&& args)
 {
     ArgumentStack stack;
     int32_t retVal = -1;
-    if (auto *pCreature = creature(args))
+
+    const auto classId = Services::Events::ExtractArgument<int32_t>(args);
+    ASSERT_OR_THROW((classId >= Constants::ClassType::MIN) && (classId <= Constants::ClassType::MAX));
+
+    CNWClass* pClass = classId < Globals::Rules()->m_nNumClasses ? &Globals::Rules()->m_lstClasses[classId] : nullptr;
+    ASSERT_OR_THROW(pClass != nullptr);
+
+    if (auto* pCreature = creature(args))
```

Comment:
You extract the arguments in the wrong order here. 

Should be
```
if (auto* pCreature = creature(args))
{ 
    const auto classId = Services::Events::ExtractArgument<int32_t>(args);
... 
```

Reply:
Good catch, copying & pasting is always a bad idea.